### PR TITLE
feat: Cross Reference Shipment in Scan Table

### PR DESCRIPTION
**Pull request by Walls2002**

### CODE REVIEW

File: src/components/warehouse/scanIncoming.tsx [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+
+function PackageScanTile({
+  switchTab,
+  packageList,
+}: {
+  switchTab: () => void
+  packageList: (data: number[]) => void
+}) {
+  const [scannedPackageIds, setScannedPackageIds] = useState<number[]>([])
+  console.log("DataFromChild", JSON.stringify(scannedPackageIds))
```
Comment (arvl130) [inline src/components/warehouse/scanIncoming.tsx]: Don't leave unnecessary `console.log`s.

File: src/components/warehouse/scanIncoming.tsx [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+function ScanTable(props: { packageList: (data: number[]) => void }) {
+  const { user, role } = useSession()
+  const [packageIds, setPackageIds] = useState<number[]>([])
+  const [isCheckedClearBtn, setIsCheckedClearBtn] = useState(false)
+  const [checkedIds, setCheckedIds] = useState<number[]>([])
+  const [SelectedChangeStatus, setSelectedChangeStatus] = useState([
+    "true",
+    "0",
+  ])
+  //const [ScannedPackagesId,setScannedPackagesId]=useState<number[]>([])
```
Comment (arvl130) [inline src/components/warehouse/scanIncoming.tsx]: Don't leave unused code.

File: src/components/warehouse/scanIncoming.tsx [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+    },
+    {
+      enabled: user !== null && role === "WAREHOUSE",
+    },
+  )
+
+  function handleChange() {
+    if (isCheckedClearBtn === false) {
+      setIsCheckedClearBtn(true)
+      console.log("CHECKED")
```
Comment (arvl130) [inline src/components/warehouse/scanIncoming.tsx]: Remove unnecessary `console.log`s.

File: src/components/warehouse/scanOutgoing.tsx [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+    {
+      enabled: user !== null && role === "WAREHOUSE",
+    },
+  )
+  const [isCheckedClearBtn, setIsCheckedClearBtn] = useState(false)
+  const [checkedIds, setCheckedIds] = useState<number[]>([])
+  function handleChange() {
+    if (isCheckedClearBtn === false) {
+      setIsCheckedClearBtn(true)
+      console.log("CHECKED")
```
Comment (arvl130) [inline src/components/warehouse/scanOutgoing.tsx]: same as above

File: src/components/warehouse/scanOutgoing.tsx [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+                        key={_package.id}
+                        package={_package}
+                        isAllChecked={isCheckedClearBtn}
+                        setCheckedId={({ id, isChecked }) => {
+                          if (isChecked === true) {
+                            setCheckedIds((CurrentCheckedIds) => {
+                              return [...CurrentCheckedIds, id]
+                            })
+                          } else {
+                            setCheckedIds((CurrentCheckedIds) => {
```
Comment (arvl130) [inline src/components/warehouse/scanOutgoing.tsx]: Variable names should be camel-cased.

File: src/components/warehouse/scanIncoming.tsx [hunk 6]
```diff
@@ -0,0 +1,640 @@
...
+    </article>
+  )
+}
+
+function ScanTable(props: { packageList: (data: number[]) => void }) {
+  const { user, role } = useSession()
+  const [packageIds, setPackageIds] = useState<number[]>([])
+  const [isCheckedClearBtn, setIsCheckedClearBtn] = useState(false)
+  const [checkedIds, setCheckedIds] = useState<number[]>([])
+  const [SelectedChangeStatus, setSelectedChangeStatus] = useState([
```
Comment (arvl130) [inline src/components/warehouse/scanIncoming.tsx]: State variables should be camel-cased

File: src/pages/warehouse/packages/scan.tsx [hunk 7]
```diff
@@ -1,352 +1,27 @@
...
-              Outgoing
-            </button>
-          </div>
-        </div>
-      </div>
-
-      <ScanTable></ScanTable>
-    </article>
+export default function ScanPage() {
+  const [selectedTabb, setSelectedtab] = useState<"Incoming" | "Outgoing">(
```
Comment (arvl130) [inline src/pages/warehouse/packages/scan.tsx]: misspelled variable name, camel-casing should be applied at word-level, i.e. `setSelectedTab` not `setSelectedtab`.

File: src/server/trpc/routers/shipment.ts [hunk 8]
```diff
@@ -1,14 +1,65 @@
...
 
 export const shipmentRouter = router({
   getAll: protectedProcedure.query(async ({ ctx }) => {
     return await ctx.db.select().from(shipments)
   }),
+  getAllInTransitStatus: protectedProcedure.query(async ({ ctx }) => {
+    const shipmentStatus = alias(shipmentStatusLogs, "shipment_inTransit")
+
+    /*const sub=ctx.db.select({shipment_id:shipmentStatusLogs.shipmentId
+    ,latest_log_timestamp:sql<Date>`MAX(${shipmentStatusLogs.createdAt})`}).from(shipmentStatusLogs).groupBy(shipmentStatusLogs.shipmentId)*/
```
Comment (arvl130) [inline src/server/trpc/routers/shipment.ts]: Remove unused/commented code.

File: src/server/trpc/routers/shipment.ts [hunk 9]
```diff
@@ -76,6 +127,85 @@ export const shipmentRouter = router({
...
+      })
+      console.log("ItemWithPackages", shipmentItemsWithPackages)
+      const items: any[] = []
+      for (const _package of shipmentItemsWithPackages[0].packages) {
+        items.push({ ..._package, status: await getLatestStatus(_package.id) })
+      }
+
+      console.log(
+        "RETURN",
+        [{ ...shipmentItemsWithPackages[0], packages: items }][0].packages,
```
Comment (arvl130) [inline src/server/trpc/routers/shipment.ts]: Remove unnecessary `console.log`s.

File: src/components/warehouse/scanIncoming.tsx [hunk 10]
```diff
@@ -0,0 +1,640 @@
...
+          className="text-center text-white bg-[#F17834] p-1"
+        >
+          <p>{_package.status.status}</p>
+        </div>
+      </td>
+    </tr>
+  )
+}
+
+const disableScanner = createContext("false")
```
Comment (arvl130) [inline src/components/warehouse/scanIncoming.tsx]: Context variable should be in Pascal case, e.g. `AuthContext`, `ThemeContext`.

File: src/components/warehouse/scanOutgoing.tsx [hunk 11]
```diff
@@ -0,0 +1,521 @@
...
+  function handleChange() {
+    if (isCheckedClearBtn === false) {
+      setIsCheckedClearBtn(true)
+      console.log("CHECKED")
+    } else {
+      setIsCheckedClearBtn(false)
+      console.log("UNCHECKED")
+    }
+  }
+  const disableScannerCont = useContext(disableScanner) === "false"
```
Comment (arvl130) [inline src/components/warehouse/scanOutgoing.tsx]: Context variables should follow Pascal case. Refer to the React [documentation](https://react.dev/reference/react/createContext) for examples.

File: src/components/warehouse/scanOutgoing.tsx [hunk 12]
```diff
@@ -0,0 +1,521 @@
...
+  isAllChecked,
+  setCheckedId,
+  checkedIds,
+}: {
+  package: Package
+  isAllChecked: boolean
+  setCheckedId: (checkedId: { id: number; isChecked: boolean }) => void
+  checkedIds: number[]
+}) {
+  const [isChecked, setisChecked] = useState(false)
```
Comment (arvl130) [inline src/components/warehouse/scanOutgoing.tsx]: Camel-casing should be applied at the word-level, e.g. `setIsChecked` not `setisChecked`.

File: src/components/warehouse/scanOutgoing.tsx [hunk 13]
```diff
@@ -0,0 +1,521 @@
...
+      enabled: user !== null && role === "WAREHOUSE",
+    },
+  )
+  /*if (isLoadingShipments || isErrorShipments) {
+      return (
+        <>
+          <LoadingSpinner></LoadingSpinner>
+        </>
+      )
+    }*/
```
Comment (arvl130) [inline src/components/warehouse/scanOutgoing.tsx]: Remove unused code.

File: src/server/trpc/routers/shipment.ts [hunk 14]
```diff
@@ -1,14 +1,65 @@
...
+      FROM shipment_status_logs 
+      GROUP BY shipment_id
+    ) AS latest_logs
+    ON shipments.id = latest_logs.shipment_id
+    JOIN shipment_status_logs
+    ON latest_logs.shipment_id = shipment_status_logs.shipment_id AND latest_logs.latest_log_timestamp = shipment_status_logs.created_at JOIN shipment_hubs ON shipments.origin_hub_id=shipment_hubs.id WHERE shipment_status_logs.status="IN_TRANSIT"`)
+
+    const [results, properties] = await ctx.db.execute(test)
+
+    /*console.log("RESULTS",results,"PROPS",properties)*/
```
Comment (arvl130) [inline src/server/trpc/routers/shipment.ts]: Remove unused code.

File: src/components/warehouse/scanIncoming.tsx [hunk 15]
```diff
@@ -0,0 +1,640 @@
...
+    },
+  )
+
+  function handleChange() {
+    if (isCheckedClearBtn === false) {
+      setIsCheckedClearBtn(true)
+      console.log("CHECKED")
+    } else {
+      setIsCheckedClearBtn(false)
+      console.log("UNCHECKED")
```
Comment (arvl130) [inline src/components/warehouse/scanIncoming.tsx]: same as above

File: src/components/warehouse/scanIncoming.tsx [hunk 16]
```diff
@@ -0,0 +1,640 @@
...
+      createdById: string
+      description: string
+      status: string
+      packageId: number
+    }
+  })[]
+}
+
+let selectedShipmentResultIds = [] as number[]
+let ScannedPackageIds = [] as number[]
```
Comment (arvl130) [inline src/components/warehouse/scanIncoming.tsx]: Use camel-case by default. Pascal case applies only to types, React components, and React context variables.

**Review by arvl130 — CHANGES_REQUESTED** [review]

File: src/components/warehouse/scanIncoming.tsx (hunk 1 above)
Comment (Walls2002) [reply src/components/warehouse/scanIncoming.tsx]: done

File: src/components/warehouse/scanIncoming.tsx (hunk 2 above)
Comment (Walls2002) [reply src/components/warehouse/scanIncoming.tsx]: done

File: src/components/warehouse/scanIncoming.tsx (hunk 3 above)
Comment (Walls2002) [reply src/components/warehouse/scanIncoming.tsx]: done

File: src/components/warehouse/scanOutgoing.tsx (hunk 4 above)
Comment (Walls2002) [reply src/components/warehouse/scanOutgoing.tsx]: done

File: src/components/warehouse/scanOutgoing.tsx (hunk 5 above)
Comment (Walls2002) [reply src/components/warehouse/scanOutgoing.tsx]: done

File: src/components/warehouse/scanIncoming.tsx (hunk 6 above)
Comment (Walls2002) [reply src/components/warehouse/scanIncoming.tsx]: done

File: src/pages/warehouse/packages/scan.tsx (hunk 7 above)
Comment (Walls2002) [reply src/pages/warehouse/packages/scan.tsx]: done

File: src/server/trpc/routers/shipment.ts (hunk 8 above)
Comment (Walls2002) [reply src/server/trpc/routers/shipment.ts]: done

File: src/server/trpc/routers/shipment.ts (hunk 9 above)
Comment (Walls2002) [reply src/server/trpc/routers/shipment.ts]: done

File: src/components/warehouse/scanIncoming.tsx (hunk 10 above)
Comment (Walls2002) [reply src/components/warehouse/scanIncoming.tsx]: done

File: src/components/warehouse/scanOutgoing.tsx (hunk 12 above)
Comment (Walls2002) [reply src/components/warehouse/scanOutgoing.tsx]: done

File: src/components/warehouse/scanOutgoing.tsx (hunk 11 above)
Comment (Walls2002) [reply src/components/warehouse/scanOutgoing.tsx]: done

File: src/components/warehouse/scanOutgoing.tsx (hunk 13 above)
Comment (Walls2002) [reply src/components/warehouse/scanOutgoing.tsx]: done

File: src/server/trpc/routers/shipment.ts (hunk 14 above)
Comment (Walls2002) [reply src/server/trpc/routers/shipment.ts]: done

File: src/components/warehouse/scanIncoming.tsx (hunk 15 above)
Comment (Walls2002) [reply src/components/warehouse/scanIncoming.tsx]: done

File: src/components/warehouse/scanIncoming.tsx (hunk 16 above)
Comment (Walls2002) [reply src/components/warehouse/scanIncoming.tsx]: done

**Review by arvl130 — APPROVED** [review]

**Pull request merged**